### PR TITLE
lr decay scheduler + save & load ckpts

### PR DESCRIPTION
**Pull request by azamatkhid**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -33,38 +33,38 @@ def __init__(self, cfg):
 
         self.test_transforms = transforms.Compose([transforms.Resize((224, 224), interpolation=2),
             transforms.ToTensor(),
-            transforms.Normalize(mean=[0.5, 0.5, 0.5],
-                std=[0.5, 0.5, 0.5])])
+            transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])])
```
Comment (flrngel) [inline app.py:36]: is this right 0.5 value? @Magauiya @azamatkhid

**Review by flrngel — APPROVED** [review]

File: app.py (hunk 1 above)
Comment (azamatkhid) [reply app.py:36]: @flrngel those mean and std values were inspired by:

https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html

Comment (Magauiya) [reply app.py:36]: @flrngel it depends on a dataset type. For toy datasets like Fashion MNIST, CIFAR should be fine, but ImageNet requires its own settings. In my opinion, **mean** and **variance** should be calculated from the dataset.

File: app.py [hunk 2]
```diff
@@ -33,38 +33,38 @@ def __init__(self, cfg):
...
             summary(self.net, (3, 224, 224))
-        pass
 
-    def train(self, criterion=nn.CrossEntropyLoss, optimizer=torch.optim.RMSprop):
-        self.net.train()
+        self.criterion = criterion()
+        self.optimizer = optimizer(self.net.parameters(), lr=self.cfg.lr_init)
+        self.scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(self.optimizer, mode='min', factor=0.5,
+                patience=1, verbose=False, threshold=0.0001, threshold_mode='abs', cooldown=0, min_lr=1e-8, eps=1e-08)
+
```
Comment (Magauiya) [inline app.py:58]: are you sure?

File: app.py [hunk 3]
```diff
@@ -128,12 +127,48 @@ def _validation(self):
                 total += labels.size(0)
                 correct += (predicted == labels).sum().item()
         self.net.train()
-        acc = correct / total
+        acc = correct / total * 100
         batch_loss = total_loss / len(self.valid_data)
         return acc, batch_loss
 
         pass
```
Comment (Magauiya) [inline app.py]:
```suggestion
```

File: app.py (hunk 2 above)
Comment (azamatkhid) [reply app.py:58]: Sure)

**Review by Magauiya — APPROVED** [review]
